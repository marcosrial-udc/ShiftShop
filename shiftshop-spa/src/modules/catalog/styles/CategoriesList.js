import {makeStyles} from '@material-ui/core';

const useStyles = makeStyles(theme => ({
    content: {
        padding: 0,
        paddingBottom: theme.spacing(1) + 'px !important'
    },
    list : {
        padding: 0,
        marginBottom: 0
    },
    emptyPlaceholder: {
        padding: theme.spacing(1),
        textAlign: 'center'
    },
    image: {
        maxWidth: '100%',
        [theme.breakpoints.down('sm')]: {maxHeight: '20rem'},
        [theme.breakpoints.up('md')]: {maxHeight: '25rem',},
    },
    emptyText: {fontStyle: 'italic'}
}));

export default useStyles;
