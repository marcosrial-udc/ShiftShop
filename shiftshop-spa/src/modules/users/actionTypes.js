export const GET_ROLES_COMPLETED = "project/users/getRolesCompleted";
export const LOGIN_COMPLETED = "project/users/loginCompleted";
export const LOGOUT = "project/users/logout";
export const GET_USERS_COMPLETED = "project/users/getUsersCompleted";
export const CLEAR_USERS_SEARCH = "project/users/clearUsersSearch";

