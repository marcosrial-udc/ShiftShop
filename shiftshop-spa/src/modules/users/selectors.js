const getModuleState = state => state.users;

export const getUser = state =>
    getModuleState(state).user;

export const isLoggedIn = state =>
    getUser(state) !== null;
