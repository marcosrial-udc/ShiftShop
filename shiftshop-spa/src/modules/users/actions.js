import * as actionTypes from './actionTypes';

export const login = (username, password) => ({
    type: actionTypes.LOGIN_COMPLETED,
    authenticatedUser: ({user: {username, password}})
});

export const logout = () => {
    return {type: actionTypes.LOGOUT};
};
