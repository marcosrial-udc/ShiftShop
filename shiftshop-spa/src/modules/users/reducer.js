import {combineReducers} from 'redux';

import * as actionTypes from './actionTypes';

const initialState = {
    roles: [],
    user: null,
    usersSearch: null,
    blockedUsersSearch: null
};

const roles = (state = initialState.roles, action) => {

    switch (action.type) {

        case actionTypes.GET_ROLES_COMPLETED:
            return action.roles;

        default:
            return state;
    }

};

const user = (state = initialState.user, action) => {

    switch (action.type) {

        case actionTypes.LOGIN_COMPLETED:
            return action.authenticatedUser.user;

        case actionTypes.LOGOUT:
            return initialState.user;

        default:
            return state;
    }

};

const usersSearch = (state = initialState.usersSearch, action) => {

    switch (action.type) {

        case actionTypes.GET_USERS_COMPLETED:
            return action.usersSearch;

        default:
            return state;
    }

};

const blockedUsersSearch = (state = initialState.blockedUsersSearch, action) => {

    switch (action.type) {

        case actionTypes.GET_BLOCKED_USERS_COMPLETED:
            return action.blockedUsersSearch;

        case actionTypes.CLEAR_BLOCKED_USERS_SEARCH:
            return initialState.blockedUsersSearch;

        default:
            return state;
    }

};


const reducer = combineReducers({
    roles,
    user,
    usersSearch,
    blockedUsersSearch
});

export default reducer;
