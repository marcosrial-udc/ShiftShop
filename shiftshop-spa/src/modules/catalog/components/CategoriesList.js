import React, {Fragment, useState} from 'react';
import {useSelector} from 'react-redux';
import {FormattedMessage} from 'react-intl';
import {Card, CardContent, CardHeader, Divider, List, Typography} from '@material-ui/core';

import useStyles from '../styles/CategoriesList';
import {emptyPlaceholder} from '../../../assets/images';

import CategoriesListItem from './CategoriesListItem';
import EditCategory from './EditCategory';

import * as selectors from '../selectors';

const CategoriesList = () => {
    const classes = useStyles();

    const [editCategory, setEditCategory] = useState(null);
    const categories = useSelector(selectors.getCategories);

    const cleanEdit = () => setEditCategory(null);

    return (
        <Card>
            <CardHeader title={<FormattedMessage id="project.catalog.CategoriesList.title"/>}/>
            <Divider/>
            <CardContent className={classes.content}>
                {categories.length > 0
                    ? <List className={classes.list}>
                        {categories.map((category) =>
                            (<Fragment key={category.id}>
                                <Divider/>
                                <CategoriesListItem category={category} edit={category => setEditCategory(category)}/>
                            </Fragment>))}
                    </List>
                    : <div className={classes.emptyPlaceholder}>
                        <img className={classes.image} src={emptyPlaceholder} alt="No items"/>
                        <Typography className={classes.emptyText} variant="subtitle2">
                            <FormattedMessage id="project.catalog.CategoriesList.emptyCategories"/>
                        </Typography>
                    </div>
                }
            </CardContent>
            {editCategory && <EditCategory category={editCategory} onClose={cleanEdit}/>}
        </Card>
    );
};

export default CategoriesList;
