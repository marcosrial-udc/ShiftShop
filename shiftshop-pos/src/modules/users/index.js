import * as actionTypes from './actionTypes';
import * as actions from './actions';
import reducer from './reducer';
import * as selectors from './selectors';

export {default as LoginPortal} from './components/LoginPortal';

export default {actionTypes, actions, reducer, selectors};
