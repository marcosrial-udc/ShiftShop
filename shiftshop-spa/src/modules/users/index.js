import * as actions from './actions';
import * as actionTypes from './actionTypes';
import reducer from './reducer';
import * as selectors from './selectors';

export {default as Role} from './constants/Role';
export {default as LoginPortal} from './components/LoginPortal';
export {default as LogOut} from './components/LogOut';
export {default as ProfileMenu} from './components/ProfileMenu';

export default {actions, actionTypes, reducer, selectors};
