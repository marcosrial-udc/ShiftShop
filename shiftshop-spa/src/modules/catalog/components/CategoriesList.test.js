import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import {Provider} from 'react-redux';
import {IntlProvider} from 'react-intl';

import {initReactIntl} from "../../../i18n";

import CategoriesList from './CategoriesList';
import {Role} from '../../users';

jest.mock('./CategoriesListItem', () => () => (<mockCategoriesListItem/>));

const {messages} = initReactIntl();

describe('CategoriesList snapshot', () => {

    const middlewares = [];
    const mockStore = configureStore(middlewares);

    const emptyCategories = {catalog: {categories: []}};
    const someCategories = {catalog: {categories: [{id: 1, name: 'CategoryA'}, {id: 2, name: 'CategoryB'}]}};

    const createCategoriesList = store => {

        return renderer.create(
            <Provider store={store}>
                <IntlProvider locale='en' messages={messages}>
                    <CategoriesList/>
                </IntlProvider>
            </Provider>
        );

    };

    test('Categories is empty', () => {

        // Initialize mockstore with empty state
        const store = mockStore(emptyCategories);

        const categoriesList = createCategoriesList(store);

        expect(categoriesList.toJSON()).toMatchSnapshot();

    });

    test('Categories has elements', () => {

        // Initialize mockstore with empty state
        const store = mockStore(someCategories);

        const categoriesList = createCategoriesList(store);

        expect(categoriesList.toJSON()).toMatchSnapshot();

    });

});
