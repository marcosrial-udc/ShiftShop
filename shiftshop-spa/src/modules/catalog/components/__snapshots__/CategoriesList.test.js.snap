// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoriesList snapshot Categories has elements 1`] = `
<div
  className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiCardHeader-root"
  >
    <div
      className="MuiCardHeader-content"
    >
      <span
        className="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
      >
        Categories list
      </span>
    </div>
  </div>
  <hr
    className="MuiDivider-root"
  />
  <div
    className="MuiCardContent-root makeStyles-content-2"
  >
    <ul
      className="MuiList-root makeStyles-list-3 MuiList-padding"
    >
      <hr
        className="MuiDivider-root"
      />
      <mockCategoriesListItem />
      <hr
        className="MuiDivider-root"
      />
      <mockCategoriesListItem />
    </ul>
  </div>
</div>
`;

exports[`CategoriesList snapshot Categories is empty 1`] = `
<div
  className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiCardHeader-root"
  >
    <div
      className="MuiCardHeader-content"
    >
      <span
        className="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
      >
        Categories list
      </span>
    </div>
  </div>
  <hr
    className="MuiDivider-root"
  />
  <div
    className="MuiCardContent-root makeStyles-content-2"
  >
    <div
      className="makeStyles-emptyPlaceholder-4"
    >
      <img
        alt="No items"
        className="makeStyles-image-5"
        src="empty.png"
      />
      <h6
        className="MuiTypography-root makeStyles-emptyText-6 MuiTypography-subtitle2"
      >
        Here will appear the added categories
      </h6>
    </div>
  </div>
</div>
`;
